data_safe_haven/commands/context.py

```diff
@@ -39,7 +42,11 @@ def show() -> None:
 @context_command_group.command()
 def available() -> None:
     """Show the available contexts."""
-    settings = ContextSettings.from_file()
+    try:
+        settings = ContextSettings.from_file()
+    except DataSafeHavenConfigError:
+        print("No context configuration file. Run `dsh context add` to create one.")
```

Comment:
Can these be `logger.critical` instead of `print`?

Reply:
If that raises an exception, I would want to avoid it as `typer.Exit` is more friendly for users.
If not, I would still like to leave that kind of thing for #1891.

Reply:
Yes, I mean we should keep the `typer.Exit` but have something like:

```python
except DataSafeHavenConfigError:
    logger.critical("No context configuration file. Run `dsh context add` to create one.")
    raise typer.Exit(code=1)
```

If critical raises an exception (I don't think it does, but worth checking) then drop to error instead.

Reply:
Done